b42_prepush_check.sh

```diff
@@ -1,22 +1,27 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-RED='\033[0;91m'
-GREEN='\033[0;32m'
-NC='\033[0m' # No Color
+# Needed for type builtin and globstar shell option
+[ -z "$BASH_VERSION" ] && { echo "Script must be run with bash!"; exit 1; }
 
-git_root=$(git rev-parse --show-toplevel)
+RED='\033[91m'
+GREEN='\033[32m'
+NC='\033[m' # No Color
+
+git_root="$(git rev-parse --show-toplevel)"
 
 summary_status=""
 
-pushd "$git_root"
+pushd -- "$git_root" >/dev/null || {  echo "Couldn't change directory!"; exit 2; }
```

Comment:
do I get it right, that we try to protect here from `$git_root` being `-n` with the `--`?
I'm a bit curios if it could be the case as `git rev-parse` should return full path, starting from the root...
but not a blocker, I'm fine with the `--` as well.

Reply:
Good catch, git rev-parse --show-toplevel does indeed return the absolute path by default (can be changed with --path-format). I'm removing the `--` for readability sake

Reply:
```suggestion
pushd "$git_root" >/dev/null || {  echo "Couldn't change directory!"; exit 2; }
```